### PR TITLE
Implement default editor for editing current message

### PR DESCRIPTION
**Pull request by therealjuanmartinez**
It seemed convenient to have the option to edit current message in an external editor - this change allows for user to specify an external editor and use ":e/:edit" in order to do so.  

This is for certain cases where more robust text editors can add value to the experience.

### CODE REVIEW

**Comment by kharvd** [comment]
Thanks for the contribution! Do you think we can use the EDITOR env variable instead?

**Comment by dbosk** [comment]
+1 on using the EDITOR env variable. [First VISUAL, then EDITOR.](https://unix.stackexchange.com/a/4861/43490) If none of them are set, go for a default or tell the user to set either.

File: gptcli/config.py [hunk 1]
```diff
@@ -22,6 +22,7 @@ class GptCliConfig:
     openai_api_key: Optional[str] = os.environ.get("OPENAI_API_KEY")
     anthropic_api_key: Optional[str] = os.environ.get("ANTHROPIC_API_KEY")
     google_api_key: Optional[str] = os.environ.get("GOOGLE_API_KEY")
+    default_editor: Optional[str] = os.environ.get("GPT_CLI_DEFAULT_EDITOR")
```
Comment (egitto) [inline gptcli/config.py:25]:
```suggestion
    default_editor: Optional[str] = os.environ.get("VISUAL") or os.environ.get("EDITOR")
```

Comment (therealjuanmartinez) [reply gptcli/config.py:25]: I'm good with EDITOR env variable - I'd lean towards a default if not specified (i.e. current behavior)

**Comment by therealjuanmartinez** [comment]
I see no problem with using an env EDITOR variable

**Comment by therealjuanmartinez** [comment]
Happy to use EDITOR env variable as suggested

File: gptcli/config.py (hunk 1 above)
Comment (therealjuanmartinez) [reply gptcli/config.py]: Same change implemented in a subsequent commit

**Pull request closed without merging**